### PR TITLE
73 public form transitions again

### PR DESCRIPTION
**Pull request by newtewt**
# Description

Updating public_form_transitions to async/await. Added more methods to switch to disabled control flow.

https://github.com/medic/angular10-migration/issues/73

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/transitions/public_form_transitions.spec.js [hunk 1]
```diff
@@ -230,16 +230,35 @@ const expectTransitions = (infodoc, ...transitions) => {
...
+    .then(([ changes ]) => {
+      ids = changes.map(change => change.id);
+    })
+    .then(() => sentinelUtils.waitForSentinelNative(ids))
+    .then(() => Promise.all([
+      utils.getDocsNative(ids),
+      sentinelUtils.getInfoDocsNative(ids),
+      utils.getDocNative('person1'),
+      utils.getDocNative('person2')
+    ]));
```
Comment (craig-landry) [inline tests/e2e/transitions/public_form_transitions.spec.js:249]: Having this variable be mutated down below makes it fragile; seeing how it is involved in asynchronous code makes that fragility scarier. I see that most of this code is preexisting functionality (and I know we are leaving most of these things on this pass) with only changes to introduce the "native" functions. Still, I think this one is worth doing an async/await conversion on. It's pretty small and most of the `then` blocks don't rely on data from the others. Please refactor this block to move from the promise-based approach to using async/await.

**Review by craig-landry — CHANGES_REQUESTED** [review]

File: tests/e2e/transitions/public_form_transitions.spec.js (hunk 1 above)
Comment (newtewt) [reply tests/e2e/transitions/public_form_transitions.spec.js:249]: I think we should chat about this one. This one registers a listener `apiUtils.getApiSmsChanges(messages)` and I'm not 100% on handling that with async/await. I had originally tried converting but ran into that and eventually flipped back to not be blocked.

Comment (craig-landry) [reply tests/e2e/transitions/public_form_transitions.spec.js:249]: Gotcha. In the interest of time, that's enough to hold off from refactoring this now.

**Review by craig-landry — APPROVED** [review]
Needs a conflict resolution, but otherwise all set. Thanks for making the updates!

**Pull request merged**